painttiming.bs

```diff
@@ -179,8 +182,8 @@ The {{PerformancePaintTiming}} interface {#sec-PerformancePaintTiming}
 * The {{PerformanceEntry/startTime}} attribute's getter must return a {{DOMHighResTimeStamp}} of when the paint occured.
 * The {{PerformanceEntry/duration}} attribute's getter must return 0.
 
-NOTE: A user agent implementing {{PerformancePaintTiming}} would need to include <code>"paint"</code> in {{PerformanceObserver/supportedEntryTypes}} for {{Window}} contexts.
-This allows developers to detect support for paint timing.
+NOTE: A user agent implementing {{PerformancePaintTiming}} would need to include <code>"paint"</code> in {{PerformanceObserver/supportedEntryTypes}} for any [=browsing context=] that supports it.
```

Comment:
It can still say Window, no? that's the only browsing context that's possible to support.

Reply:
"browsing context" is a more precise term, it's a window that holds a document.

Reply:
Both "[browsing context](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context)" and "[Window](https://html.spec.whatwg.org/multipage/window-object.html#window)" are precise terms. None of them is ideal here.

`supportedEntryTypes` is a frozen array on the global object. So maybe:
"would need to include 'paint' in supportedEntryTypes for global objects which browsing context supports it"
(assuming we will hang support for paint timing on browsing contexts)

Reply:
Actually, I don't think that works.
So maybe: "... would need to include 'paint' in supportedEntryTypes of the current global object, if the current global object's environment settings object's responsible Document's browsing context supports it""

Reply:
No, Windows are global objects, so Window is correct. Also not clear to me why we'd define support in terms of browsing context instead of in terms of Windows/global objects.

Reply:
(See https://html.spec.whatwg.org/multipage/webappapis.html#global-object for global object)

Reply:
> No, Windows are global objects, so Window is correct. Also not clear to me why we'd define support in terms of browsing context instead of in terms of Windows/global objects.

From my understanding, `Window` is the API to access a browsing context, and may represent an actual browsing context or null, rather than the browsing context itself.

OTOH, Global object can also be a worker global object, so it doesn’t fit with the paint-API spec.

So a browsing context is equivalent to a global object that has an associated document and window object, while a window may be associated with a null browsing context.

The reason why I went with `browsing context`, is that that’s where “nested” and “top level” are defined as ways to discern between the main frame and iframes... not sure how to address those terms with either `Window` or `global object`.

If we don’t see things the same way, maybe you can offer an alternate patch? I really don’t mind the wording as much as I would like to see the issues in the spec resolved!

Reply:
Please have a look at https://github.com/w3c/paint-timing/pull/79/files#r399782024

I had to use "browsing context" as it seems to be the standard way to separate between main frame and iframe, and I tried to use global object wherever possible, and an existing term "frozen array of entry types". WDYT?

Reply:
btw see in https://html.spec.whatwg.org/multipage/window-object.html#barprop:

If this BarProp object's relevant global object's browsing context is null, then return false.

Can't we use the same terminology here?